docs/guide/in-depth/bundling-cjs.md

```diff
@@ -91,7 +91,35 @@ The `__toESM` helper ensures that CommonJS exports are properly converted to ES
 
 ## `require` external modules
 
-Rolldown will leave it as it is. You need to ensure the running environment will provide a `require` function.
+When [`platform: 'node'`](../features.md#platform-presets) is set, Rolldown will generate a `require` function from [`module.createRequire`](https://nodejs.org/docs/latest/api/module.html#modulecreaterequirefilename).
+
+For other platforms, Rolldown will leave it as it is, so ensure that the running environment provides a `require` function or inject one manually.
```

Comment:
It might be not a practical advice in some cases.

Require polyfill has to be sync but ESM-native environments can only have it async unless there is static import or bundled (lazy loaded) modules that require polyfill returns. It is impractical as js only polyfill.

Reply:
Yeah, if you cannot import all the externalized modules statically, the example below won't work. In that case, a plugin is needed (at least for now). Probably that plugin should have [`requireReturnsDefault`](https://github.com/rollup/plugins/blob/master/packages/commonjs/README.md#requirereturnsdefault) option to control the behavior.

Reply:
I'm proposing to introduce a new rolldown plugin. Extracting this part should work similar to `requireReturnsDefault=false`.
https://github.com/vitejs/rolldown-vite/blob/52af0fcefd10475e13c5c2536dfada3561c5542d/packages/vite/src/node/optimizer/rolldownDepPlugin.ts#L317-L367

Reply:
Thnx for link! It would be also useful for Nitro vite support if extracted!